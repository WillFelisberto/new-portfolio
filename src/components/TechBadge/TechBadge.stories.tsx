import { Meta, StoryObj } from '@storybook/react';
import TechBadge from '.';

const meta: Meta<typeof TechBadge> = {
  title: 'Components/TechBadge',

  component: TechBadge,
} satisfies Meta<typeof TechBadge>;

export default meta;
type Story = StoryObj<typeof TechBadge>;

export const Default: Story = {
  args: {
    name: 'React',
    //ðŸ‘‡ The args you need here will depend on your component
  },
};
