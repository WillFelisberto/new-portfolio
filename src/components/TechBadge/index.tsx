'use client';
import { motion } from 'motion/react';
import { ComponentProps } from 'react';
type TechBadgeProps = ComponentProps<typeof motion.span> & {
  name: string;
};

const TechBadge = ({ name, ...props }: TechBadgeProps) => {
  return (
    <motion.span
      className="rounded-lg bg-sky-900/80 px-3 py-1 text-sm text-sky-400"
      {...props}
    >
      {name}
    </motion.span>
  );
};

export default TechBadge;
