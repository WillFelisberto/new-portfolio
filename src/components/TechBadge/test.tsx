import { render } from '@testing-library/react';

import TechBadge from '.';

describe('<TechBadge />', () => {
  // Renders badge with provided name prop
  it('should render badge with provided name text', () => {
    const { container, getByText } = render(<TechBadge name="React" />);

    expect(getByText('React')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render empty badge when name prop is empty string', () => {
    const { container } = render(<TechBadge name="" />);

    const badge = container.querySelector('span');
    expect(badge).toBeInTheDocument();
    expect(badge).toHaveTextContent('');
  });
});
