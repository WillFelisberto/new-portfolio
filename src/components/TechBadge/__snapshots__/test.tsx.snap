// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TechBadge /> should render badge with provided name text 1`] = `
<span
  class="rounded-lg bg-sky-900/80 px-3 py-1 text-sm text-sky-400"
>
  React
</span>
`;
